xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererVAAPIGL.cpp

```diff
@@ -77,7 +77,9 @@ bool CRendererVAAPI::Configure(const VideoPicture &picture, float fps, unsigned
 
   for (auto &tex : m_vaapiTextures)
   {
-    tex.Init(interop);
+    // FIXME How do we know which texture type to create without running the interop test every time?
+    tex.reset(VAAPI::CVaapiTexture::CreateTexture(CRendererVAAPI::m_pWinSystem->GetVADisplay(), CRendererVAAPI::m_pWinSystem->GetEGLDisplay()));
```

Comment:
don't call a static. here is the location where we either instantiate one class or the other.

Reply:
Yeah I don't really like it like this either (thus the FIXME), but do you have a suggestion about who should decide which texture interop to create? I don't think it concerns the renderer directly (as long as the textures are mapped in GL, it doesn't matter how) so it shouldn't do it itself.

Reply:
CRendererVAAPI takes the decision here. There can be non-EGL mappings like the old putSurface for GLX.

Reply:
Sure, but for the problem at hand both VAAPI1 and VAAPI2 map to GL, so the renderer does not care which it is. It doesn't need to do anything differently.

Reply:
putSurface for GLX would also map to GL. But this class certainly won't reside in a file named EGL. here is a good location.